src/Symfony/Component/HttpClient/Internal/AmpClientState.php

```diff
@@ -60,7 +61,7 @@ public function __construct(
         $this->clientConfigurator = $clientConfigurator(...);
     }
 
-    public function request(array $options, Request $request, Cancellation $cancellation, array &$info, \Closure $onProgress, &$handle): Response
+    public function request(array $options, Request $request, Cancellation $cancellation, array &$info, \Closure $onProgress, &$handle, ?DeferredCancellation $canceller = null): Response
```

Comment:
Can't we do this? I prefer not having optional args in internal methods, so that we don't forget to update a call somewhere. At least if possible. But: see my previous comment on the call to request()

```suggestion
    public function request(array $options, Request $request, Cancellation $cancellation, array &$info, \Closure $onProgress, &$handle, DeferredCancellation $canceller): Response
```

Reply:
Yes, better change `Cancellation` to `DeferredCancellation` indeed